sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExtractValue.scala

```diff
@@ -195,6 +240,20 @@ case class GetArrayItem(child: Expression, ordinal: Expression)
       baseValue(index)
     }
   }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val castToInt = if (ordinal.dataType == LongType) "(int)" else ""
```

Comment:
In order to support other numeric types, I'd like to cast it into `int` when `ordinal.dataType` is not `IntegerType`.

Btw, maybe we could always cast it into `int`, the Java compiler and JIT may could avoid the unnecessary casting for us.